salt/utils/dns.py

```diff
@@ -1,18 +1,787 @@
+    :param name: Name of record to search
+    :param rdtype: DNS record type
+    :param timeout: server response timeout
+    :return: [] of records or False if error
+    '''
+    cmd = 'dig +search +fail +noall +answer +noclass +nottl -t {0} '.format(rdtype)
+    if servers:
+        cmd += ''.join(['@{0} '.format(srv) for srv in servers])
+    if timeout is not None:
+        if servers:
+            timeout = int(float(timeout) / len(servers))
+        else:
+            timeout = int(timeout)
+        cmd += '+time={0} '.format(timeout)
+    if secure:
+        cmd += '+dnssec +adflag '
+
+    cmd = __salt__['cmd.run_all'](cmd + str(name), python_shell=False, output_loglevel='quiet')
+
+    # In this case, 0 is not the same as False
+    if cmd['retcode'] != 0:
+        log.warning(
+            'dig returned ({0}): {1}'.format(
+                cmd['retcode'], cmd['stderr']
+        ))
+        return False
+
+    validated = False
+    res = []
+    for line in cmd['stdout'].splitlines():
+        rname, rtype, rdata = line.split(None, 2)
+        if rtype == 'CNAME' and rdtype != 'CNAME':
+            continue
+        elif rtype == 'RRSIG':
+            validated = True
+            continue
+        res.append(rdata.strip(string.whitespace + '"'))
+
+    if res and secure and not validated:
+        return False
+    else:
+        return res
+
+
+def _query_drill(name, rdtype, timeout=None, servers=[], secure=None):
+    '''
+    Use drill to lookup addresses
+    :param name: Name of record to search
+    :param rdtype: DNS record type
+    :param timeout: command return timeout
+    :return: [] of records or False if error
+    '''
+    cmd = 'drill '
+    if secure:
+        cmd += '-D -o ad '
+    cmd += '{0} {1} '.format(rdtype, name)
+    if servers:
+        cmd += ''.join(['@{0} '.format(srv) for srv in servers])
+    cmd = __salt__['cmd.run_all'](
+        cmd, timeout=timeout,
+        python_shell=False, output_loglevel='quiet')
+
+    # In this case, 0 is not the same as False
+    if cmd['retcode'] != 0:
+        log.warning(
+            'drill returned ({0}): {1}'.format(
+                cmd['retcode'], cmd['stderr']
+            ))
+        return False
+
+    lookup_res = iter(cmd['stdout'].splitlines())
+    res = []
+    try:
+        line = ''
+        while 'ANSWER SECTION' not in line:
+            line = next(lookup_res)
+        while True:
+            line = next(lookup_res)
+            line = line.strip()
+            if not line or line.startswith(';;'):
+                break
+
+            l_type, l_rec = line.split(None, 4)[-2:]
+            if l_type == 'CNAME' and rdtype != 'CNAME':
+                continue
+            elif l_type == 'RRSIG':
+                validated = True
+                continue
+
+            res.append(l_rec.strip(string.whitespace + '"'))
+
+    except StopIteration:
+        pass
+    finally:
+        if res and secure and not validated:
+            return False
```

Comment:
``return`` statements should not be used in a ``finally`` block. They can mask actual exceptions. So if anything other thatn a ``StopIteration`` is raised it would be lost.

If you're confident that only ``StopIteration`` would be raised, then you should just get rid of the ``finally`` block and dedent the stuff within it.